src/PlayAstWithArkouda.py

```diff
@@ -11,87 +11,187 @@
 
 class ArkoudaVisitor(ast.NodeVisitor):
 
+    # initialize return string which will contain scheme/lisp code
     def __init__(self):
         self.ret = ""
+        self.arg_list = []
 
+    # Binary Operators
     def visit_Add(self, node):
         self.ret += " +"
 
+    def visit_Sub(self, node):
+        self.ret += " -"
+
     def visit_Mult(self, node):
         self.ret += " *"
         
+    # Comparison operators
+    def visit_Eq(self, node):
+        self.ret += " =="
+
+    def visit_NotEq(self, node):
+        self.ret += " !="
+
+    def visit_Lt(self, node):
+        self.ret += " <"
+
+    def visit_LtE(self, node):
+        self.ret += " <="
+
+    def visit_Gt(self, node):
+        self.ret += " >"
+
+    def visit_GtE(self, node):
+        self.ret += " <="
+
+    # Boolean Operators
+    def visit_And(self, node):
+        self.ret += " and"
+
+    def visit_Or(self, node):
+        self.ret += " or"
+
+    # Uniary Operators
+    def visit_Not(self, node):
+        self.ret += " not"
+
+    # Unary Op (only `not` at this point)
+    def visit_UnaryOp(self, node):
+        self.ret += " ("
+        self.visit(node.op) 
+        self.visit(node.operand)
+        self.ret += " )"
+        print(self.ret)
+
+    # Compare Op
+    def visit_Compare(self, node):
+        self.ret += " ("
+        self.visit(node.ops[0]) 
+        self.visit(node.left)  # left?
+        self.visit(node.comparators[0]) # right?
+        self.ret += " )"
+        print(self.ret)
+
+    # Boolean Op 
+    def visit_BoolOp(self, node):
+        self.ret += " ("
+        self.visit(node.op)
+        self.visit(node.values[0]) # left?
+        self.visit(node.values[1]) # right?
+        self.ret += " )"
+        print(self.ret)
+
+    # Binary Op 
     def visit_BinOp(self, node):
         self.ret += " ("
         self.visit(node.op)
         self.visit(node.left)
         self.visit(node.right)
         self.ret += " )"
         print(self.ret)
-            
+
+    # If Expression (body) if (test) else (orelse)
+    def visit_IfExp(self, node):
+        self.ret += " ( if"
+        self.visit(node.test)
+        self.visit(node.body)
+        self.visit(node.orelse)
+        self.ret += " )"
+        print(self.ret)
+
+    # Assignment which returns a value `(x := 5)`
```

Comment:
Nit: isn't this expressed as `(:= x 5)`, not `(x := 5)`?

Reply:
in Python it is `(x := 5)` in scheme/lisp it is `(:= x 5)`, this node of the `ast` a `ast.NamedExpr` represents the Python expression. we are walking the python `ast` and generating scheme/lisp.